### PR TITLE
fix: fetch segments to prevent long running query when segmenting

### PR DESCRIPTION
**Pull request by almeidabbm**
### Description:

Fetches top `MAX_SEGMENT_DIMENSION_UNIQUE_VALUES` segments by metric before running the query so that a filter can be applied to reduce amount of results.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts [hunk 1]
```diff
@@ -254,6 +256,36 @@ export class MetricsExplorerService<
...
+    ) {
+        if (!segmentDimension) {
+            return [];
+        }
+
+        const getSegmentsMetricQuery: MetricQuery = {
+            ...metricQuery,
+            exploreName,
+            dimensions: [segmentDimension],
+            sorts: [{ fieldId: segmentDimension, descending: true }],
```
Comment (IrakliJani) [inline packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts:274]: I believe the sort here should be by a metric

**Review by IrakliJani — CHANGES_REQUESTED** [review]
just one requested change

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts]: Yep, that's true, changed in most recent commit! Thank you 🙏

**Review by IrakliJani — APPROVED** [review]
nice!

**Pull request merged**